docs/cvl/methods.md

```diff
@@ -39,7 +39,10 @@ methods          ::= "methods" "{" { method_spec } "}"
 
 method_spec      ::= ( sighash | [ id "." ] id "(" evm_params ")" )
                      [ "returns" types ]
-                     [ "envfree" ]
+                     { | "with" "(" "env" id ")"
+                       | "receiver" "(" "address" id ")"
```

Comment:
I think the receiver is a bad idea:
1. it can be confused with the notion of receive function in solidity
2. making “receiver” a keyword in cvl will be horrible as "receiver" is a common name used in many defi applications and I assume in many specs receiver has been used as a variable name.  
3. we have the currentContract keyword so why not have something similar? I’m ok with other names instead of 'calledContract'  but not new syntax style unless there’s a good reason

Reply:
the main reason is the difference between `with(env)` and `calledContract`.  Perhaps we could use `calledContract` and `calledEnv` or something?

Reply:
also, there's no reason we can't continue to use `receiver` as a variable or function name and also have it in this context, although I agree that it's probably not the best choice

Reply:
I also thing the `receiver` is a little too much - after all it can only hold a parameter of a predetermined type, right? (this is something weird also about the `with` clause, but we're not going to change that now...).

The `calledEnv` idea is nice, I just wonder if it would be confusing for users because of the existing preserved `with` clause.

Reply:
I updated to keep `with` clause and `calledContract`.